TrackletAlgorithm/MatchProcessor.h

```diff
@@ -220,85 +258,612 @@ ap_uint<width> iabs( ap_int<width> value )
+#  include "../emData/MP/tables/MP_ProjectionBend_D3.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else if (L==TF::D4){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_ProjectionBend_D4.tab")
+#  include "../emData/MP/tables/MP_ProjectionBend_D4.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else if (L==TF::D5){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_ProjectionBend_D5.tab")
+#  include "../emData/MP/tables/MP_ProjectionBend_D5.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else {
+      static_assert(true, "Only DISKS for rinv bend");
+    }
+  }
+} // end readTable_rbend()
+
+template<MC::lutType type, TF::layerDisk L, int width, int depth>
+void readTable_rDSS(ap_uint<width> table[depth]){
+  if(type==MC::RDSSINNERCUT) { // rDSSinner cuts (//for disks only)
+    if (L==TF::D1){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_D1PHIC_rDSSinner.tab")
+#  include "../emData/MP/tables/MP_D1PHIC_rDSSinner.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else if (L==TF::D2){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_D2PHIC_rDSSinner.tab")
+#  include "../emData/MP/tables/MP_D2PHIC_rDSSinner.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else if (L==TF::D3){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_D3PHIC_rDSSinner.tab")
+#  include "../emData/MP/tables/MP_D3PHIC_rDSSinner.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else {
+      static_assert(true, "Only DISKS 1 and 2 are valid for rDSS inner");
+    }
+  }
+  else if(type==MC::RDSSOUTERCUT) { // rDSSouter cuts (//for disks only)
+    if (L==TF::D3){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_D3PHIC_rDSSouter.tab")
+#  include "../emData/MP/tables/MP_D3PHIC_rDSSouter.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else if (L==TF::D4){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_D4PHIC_rDSSouter.tab")
+#  include "../emData/MP/tables/MP_D4PHIC_rDSSouter.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else if (L==TF::D5){
+      ap_uint<width> tmp[depth] =
+#if __has_include("../emData/MP/tables/MP_D5PHIC_rDSSouter.tab")
+#  include "../emData/MP/tables/MP_D5PHIC_rDSSouter.tab"
+#else
+      {};
+#endif
+      for (int i = 0; i < depth; i++) table[i] = tmp[i];
+    }
+    else {
+      static_assert(true, "Only DISKS 3 to 5 are valid for rDSS outer");
+    }
+  }
+} // end readTable_rDSS()
```

Comment:
This code above to read all these tables is rather repetitive - is there a way to simplify this? I think this applies not only to the MP, but other modules as well. Maybe this should not hold up this PR, but we should create an issue for this?

Reply:
I agree. I'll think a bit about this and open a new PR.